microservices/realtime/pkg/services/feeder/call.go

```diff
@@ -0,0 +1,121 @@
+package feeder
+
+import (
+	"errors"
+	"time"
+)
+
+// Call is an abstract class representing a call.
+type AbstractCall struct {
+	ID      string
+	CallID  string
+	Status  string
+	Channel string
+}
+
+// Call statuses
+const (
+	CALL_SETUP  = "Trying"
+	RINGING     = "Early"
+	IN_CALL     = "Confirmed"
+	HANG_UP     = "Terminated"
+	UPDATE_CLID = "UpdateCLID"
+)
+
+// SignificantCallEvents is an array of significant call events.
+var SignificantCallEvents = []string{CALL_SETUP, RINGING, IN_CALL, HANG_UP, UPDATE_CLID}
+
+// IsDone returns true if the call status is HangUp.
+func (c *AbstractCall) IsDone() bool {
+	return c.Status == HANG_UP
+}
+
+// GetCallID returns the call ID.
+func (c *AbstractCall) GetCallID() string {
+	return c.CallID
+}
+
+// GetStatus returns the call status.
+func (c *AbstractCall) GetStatus() string {
+	return c.Status
+}
+
+// GetChannel returns the call channel.
+func (c *AbstractCall) GetChannel() string {
+	return c.Channel
+}
```

Comment:
If you add getters for struct fields, it's probably a good idea to make fields private to enforce the usage of these getters